fxa-oauth-server/lib/routes/token.js

```diff
@@ -344,39 +355,41 @@ function confirmCode(id, code) {
   });
 }
 
-function confirmRefreshToken(params) {
-  return db.getRefreshToken(encrypt.hash(params.refresh_token))
-  .then(function(tokObj) {
-    if (! tokObj) {
-      logger.debug('refresh_token.notFound', params.refresh_token);
-      throw AppError.invalidToken();
-    } else if (hex(tokObj.clientId) !== hex(params.client_id)) {
-      logger.debug('refresh_token.mismatch', {
-        client: params.client_id,
-        code: tokObj.clientId
-      });
-      throw AppError.invalidToken();
-    } else if (! tokObj.scope.contains(params.scope)) {
-      logger.debug('refresh_token.invalidScopes', {
-        allowed: tokObj.scope,
-        requested: params.scope
-      });
-      throw AppError.invalidScopes();
-    }
-    tokObj.scope = params.scope;
+async function confirmRefreshToken(params) {
+  const tokObj = await db.getRefreshToken(encrypt.hash(params.refresh_token));
+  if (! tokObj) {
+    logger.debug('refresh_token.notFound', params.refresh_token);
+    throw AppError.invalidToken();
+  } else if (hex(tokObj.clientId) !== hex(params.client_id)) {
+    logger.debug('refresh_token.mismatch', {
+      client: params.client_id,
+      code: tokObj.clientId
+    });
+    throw AppError.invalidToken();
+  } else if (! tokObj.scope.contains(params.scope)) {
+    logger.debug('refresh_token.invalidScopes', {
+      allowed: tokObj.scope,
+      requested: params.scope
+    });
+    throw AppError.invalidScopes();
+  }
+  tokObj.scope = params.scope;
 
-    var now = new Date();
-    var lastUsedAt = tokObj.lastUsedAt;
+  var now = new Date();
+  var lastUsedAt = tokObj.lastUsedAt;
 
-    if ((now - lastUsedAt) > REFRESH_LAST_USED_AT_UPDATE_AFTER_MS){
-      db.usedRefreshToken(encrypt.hash(params.refresh_token)).then(function() {
-        logger.debug('usedRefreshToken.updated', now);
-      });
-    } else {
-      logger.debug('usedRefreshToken.not_updated');
-    }
-    return tokObj;
-  });
+  if (params.instance_id && tokObj.instanceId !== params.instance_id) {
+    await db.updateRefreshTokenInstanceId(encrypt.hash(params.refresh_token), buf(params.instance_id));
+    tokObj.instanceId = params.instance_id;
+  }
```

Comment:
I'm not convinced we need to accept `instance_id` in params here, and the fact that we need to add logic for changing the `instance_id` associated with an existing refresh token makes me even less convinced.  In a world where clients only hold a single refresh token and can use incremental auth to update it, would we need this?

Perhaps another way of asking is: how bad would it be if we delayed this "let the client explicitly pass around its `instance_id`" logic until we've got the core data model up and running and have some experience with using it from the client's perspective?